e2e/e2etests/test_spl_withdraw_and_call_revert.go

```diff
@@ -0,0 +1,101 @@
+package e2etests
```

Comment:
Can we maybe have one of the two tests using a smart contract with onRevert? Add don't think we need two more tests but adding one for one of the existing test to add more coverage, onRevert hook used for Solana CCTX